### ISSUE TITLE
Tests with @example and @env run regardless of the given --env

### ISSUE THREAD

**Issue by gfrooo1989** [Codeception/Codeception#3502]
#### What are you trying to achieve?

Run environment specific tests that have multiple @example
#### What do you get instead?

Tests are run everytime regardless of --env option if they contain the @example

> Provide console output if related. Use `-vvv` mode for more details.
> php codecept.phar run --env uat3

✔ ViewCest: Check product price | input_is_here (0.51s)
✔ ViewCest: Check product price | input_is_here (0.57s)
✔ ViewCest: Check product price | input_is_here (0.57s)
✔ ViewCest: Check product price | input_is_here (0.58s)
✔ ViewCest: Check product price | input_is_here (0.54s)
✔ ViewCest: Check product price | input_is_here (0.59s)
✔ LoginCest: Check successful login (3.25s)
✔ LoginCest: Check unsuccessful login (1.62s)
✔ RegisterCest: Check successful register (2.85s)
✔ RegisterCest: Check unsuccessful register (2.79s)

> Provide test source code if related

``` php
    /**
     * 
     * @param AcceptanceTester $I
     * @param \Page\product\Details $product
     * 
     * @example [input array here]
     * @example [input array here]
     * @example [input array here]
     * 
     * @example [input array here]
     * @example [input array here]
     * @example [input array here]
     *
     * @env uat1
     * 
     * @after logout
     */
```
### Details
- Codeception version: 2.2.4
- PHP Version: 5.5.9
- Operating System: Ubuntu 14.04 x64
- Installation type: Composer
- Suite configuration:
  Multiple env specific settings... nothing fancy here

---

So, I am trying to run all tests that belong to different ENV, everything works good for some 50 tests, but just for this test I have a problem. It runs every time regardless of the input given via --env. The difference between this and the other tests is that this test contains the @example .

**Comment by mvadzim** [comment]
try change in https://github.com/Codeception/Codeception/blob/2.2/src/Codeception/SuiteManager.php#L112

```
if ($test instanceof \PHPUnit_Framework_TestSuite_DataProvider) {
    foreach ($test->tests() as $t) {
        $this->configureTest($t);
    }
}
```

to 

```
if ($test instanceof \PHPUnit_Framework_TestSuite_DataProvider) {
    foreach ($test->tests() as $t) {
        $this->configureTest($t);
        if (!$this->isExecutedInCurrentEnvironment($t)) {
            return; // skip tests from other environments
        }
    }
}
```

**Comment by gfrooo1989** [comment]
Thanks for this piece of code. This fixes my problems, imo it should be included in a future release.

**Comment by edno** [comment]
@mvadzim Can you do a PR for this fix?

**Comment by michaeldoehler** [comment]
Seems to be fixed in current version 2.2 or? This issue can be closed?!

**Comment by Naktibalda** [comment]
Probably it was fixed by @edno with #3863

**Comment by edno** [comment]
I confirmed that is fixed with current version. I had previously the same issue in one of my testset.
